src/genie/libs/parser/iosxe/show_ap.py

```diff
@@ -0,0 +1,130 @@
+            }
+        }
+    }
+
+
+# ====================================
+# Parser for:
+#  * 'show ap dot11 dual-band summary'
+# ====================================
+class ShowApDot11DualBandSummary(ShowApDot11DualBandSummarySchema):
+    """Parser for show ap dot11 dual-band summary"""
+
+    cli_command = 'show ap dot11 dual-band summary'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        show_ap_dot11_dual_band_summary_dict = {}
+
+        # AP Name                           Mac Address     Slot  Admin State   Oper State     Width  Txpwr           Mode    Subband    channel
+        # ---------------------------------------------------------------------------------------------------------------------------------------------------------
+        # aa-test-4800                 64d8.14ec.1120  0     Enabled       Down           20     *1/8 (23 dBm)   Local   All        (6)*
+        # aa-test-4800                 64d8.14ec.1120  2     Enabled       Down           20     N/A             Monitor All        N/A
+
+        remove_lines = ()
+
+        # Remove unwanted lines from raw text
+        def filter_lines(raw_output, remove_lines):
+            # Remove empty lines
+            clean_lines = list(filter(None, raw_output.splitlines()))
+            rendered_lines = []
+            for clean_line in clean_lines:
+                clean_line_strip = clean_line.strip()
+                # Remove lines unwanted lines from list of "remove_lines"
+                if not clean_line_strip.startswith(remove_lines):
+                    rendered_lines.append(clean_line_strip)
+            return rendered_lines
+
+        out = filter_lines(raw_output=out, remove_lines=remove_lines)
+
+        # AP Name                           Mac Address     Slot  Admin State   Oper State     Width  Txpwr           Mode    Subband    channel
+        ap_info_headers_capture = re.compile(
+            r"^AP\s+Name\s+Mac\s+Address\s+Slot\s+Admin\s+State\s+Oper\s+State\s+Width\s+Txpwr\s+Mode\s+Subband\s+channel$")
+        # ---------------------------------------------------------------------------------------------------------------------------------------------------------
+        delimiter_capture = re.compile(
+            r"^---------------------------------------------------------------------------------------------------------------------------------------------------------$")
+        # aa-test-4800                 64d8.14ec.1120  0     Enabled       Down           20     *1/8 (23 dBm)   Local   All        (6)*
+        ap_info_capture = re.compile(
+            r"^(?P<ap_name>\S+)\s+(?P<ap_mac_address>\S+)\s+(?P<slot_id>\d+)\s+(?P<admin_state>(Enabled|Disabled))\s+(?P<oper_state>\S+)\s+(?P<width>\d+)\s+(?P<tx_pwr>(N\/A|\*.*m\)))\s+(?P<mode>\S+)\s+(?P<subband>\S+)\s+(?P<channel>\S+)$")
+
+        ap_index = 0
+
+        for line in out:
```

Comment:
This for loop can be a lot smaller, please see example:
```
ret_dict = {}

for line in out:

	# aa-test-4800                 64d8.14ec.1120  0     Enabled       Down           20     *1/8 (23 dBm)   Local   All        (6)*
	m = ap_info_capture.match(line)
	if m:
		groups = m.groupdict()
		ap_index += 1
		
		ap_index_dict = ret_dict.setdefault('ap_dot11_dual-band_summary', {}).\
			setdefault(ap_index, {})
		
		ap_index_dict.update('ap_name': groups['ap_name'])
		ap_index_dict.update('ap_mac_address': groups['ap_mac_address'])
		ap_index_dict.update('slot_id': int(groups['slot_id']))
		ap_index_dict.update('admin_state': groups['admin_state'])
		ap_index_dict.update('oper_state': groups['oper_state'])
		ap_index_dict.update('width': int(groups['width']))
		ap_index_dict.update('tx_pwr': groups['tx_pwr'])
		ap_index_dict.update('mode': groups['mode'])
		ap_index_dict.update('subband': groups['subband'])
		ap_index_dict.update('channel': groups['channel'])
		
return ret_dict
```

This also allows us to remove both `ap_info_headers_capture` and `delimiter_capture` regex. We dont have to match any lines we arent actively using.

If you change schema based on @tahigash comment then its also simple as changing:
```
		ap_index_dict = ret_dict.setdefault('ap_dot11_dual-band_summary', {}).\
			setdefault(ap_index, {})
```
to
```
		ap_index_dict = ret_dict.setdefault('ap_dot11_dual-band_summary', {}).\
			setdefault('indexes', {}).setdefault(ap_index, {})
```